src/v/redpanda/request_auth.h

```diff
@@ -73,9 +76,11 @@ class [[nodiscard]] request_auth_result {
     void pass();
 
     ss::sstring const& get_username() const { return _username; }
+    ss::sstring const& get_password() const { return _password; }
 
 private:
     security::credential_user _username;
+    ss::sstring _password;
```

Comment:
Should we store an encrypted version of the password instead? This change is necessary because the proxy will need the pass to create authenticated connections to the Kafka API.

cc: @dotnwat @jcsp

Reply:
Not necessary to encrypt it in-memory (just be careful not to log it).